src/Haskell/Ide/Engine/Plugin/HieExtras.hs

```diff
@@ -245,33 +262,48 @@ instance ModuleCache CachedCompletions where
       { allModNamesAsNS = allModNamesAsNS
       , unqualCompls = toplevelCompls ++ unquals
       , qualCompls = quals
+      , importableModules = moduleNames
       }
 
-getCompletions :: Uri -> (T.Text, T.Text) -> IdeM (IdeResponse [J.CompletionItem])
-getCompletions uri (qualifier, ident) = pluginGetFileResponse "getCompletions: " uri $ \file ->
+getCompletions :: Uri -> PosPrefixInfo -> IdeM (IdeResponse [J.CompletionItem])
+getCompletions uri prefixInfo = pluginGetFileResponse "getCompletions: " uri $ \file ->
   let handlers =
         [ GM.GHandler $ \(ex :: SomeException) ->
             return $ IdeResponseFail $ IdeError PluginError
-                                                (T.pack $ "getCompletions" <> ": " <> (show ex))
+                                                (T.pack $ "getCompletions" <> ": " <> show ex)
                                                 Null
         ]
+      PosPrefixInfo {fullLine, prefixModule, prefixText} = prefixInfo
   in flip GM.gcatches handlers $ do
-    -- debugm $ "got prefix" ++ show (qualifier, ident)
-    let enteredQual = if T.null qualifier then "" else qualifier <> "."
-        fullPrefix = enteredQual <> ident
-    withCachedModuleAndData file $ \_ CC { allModNamesAsNS, unqualCompls, qualCompls } ->
+    debugm $ "got prefix" ++ show (prefixModule, prefixText)
+    let enteredQual = if T.null prefixModule then "" else prefixModule <> "."
+        fullPrefix = enteredQual <> prefixText
+    withCachedModuleAndData file $ \_ CC { allModNamesAsNS, unqualCompls, qualCompls, importableModules } ->
       let
         filtModNameCompls = map mkModCompl
           $ mapMaybe (T.stripPrefix enteredQual)
           $ Fuzzy.simpleFilter fullPrefix allModNamesAsNS
 
-        filtCompls = Fuzzy.filterBy label ident compls
+        filtCompls = Fuzzy.filterBy label prefixText compls
           where
-            compls = if T.null qualifier
+           compls = if T.null prefixModule
               then unqualCompls
-              else Map.findWithDefault [] qualifier qualCompls
+              else Map.findWithDefault [] prefixModule qualCompls
+
+        insertWithoutPrefix orig = J.detail ?~ orig
```

Comment:
I'm not sure why this line requires `FlexibleContexts`

Reply:
Would adding a type signature to `insertWithoutPrefix` alleviate the need for it?